harmony/extensions/game.py

```diff
@@ -35,6 +57,63 @@ def __init__(self, bot):
             extra = ' You said: {}'.format(words)
         await self.bot.say('You\'re {}.{}'.format(ctx.message.author, extra))
 
+    @commands.command(pass_context=True)
+    async def makechar(self, ctx):
+        session = self.bot.Session()
+        try:
+            player = session.query(Player).filter(Player.user_id == ctx.message.author.id)[0]
+            status = 'existing'
+        except IndexError:
+            player = Player(user_id=ctx.message.author.id, user_name=ctx.message.author.display_name)
+            status = 'new'
+
+        session.add(player)
+        session.commit()
+        await self.bot.say('```\n{}\'s {} player:\nTOUGHNESS: {}\nINTELLIGENCE: {}\nTACT: {}\nSPEED: {}```'.format(
+            player.user_name,
+            status,
+            player.toughness,
+            player.intelligence,
+            player.tact,
+            player.speed
+        ))
+
+    @commands.command(pass_context=True)
+    async def battle(self, ctx):
+        session = self.bot.Session()
+        try:
+            player = session.query(Player).filter(Player.user_id == ctx.message.author.id)[0]
+        except IndexError:
+            await self.bot.say('You don\'t have a character. Make one with `!makechar`.')
+            return
+
+        if player.hit_points <= 0 or player.enemy_hit_points <= 0:
+            player.reset()
+            if player.user_id in self.battles:
+                del self.battles[player.user_id]
+
+        attack = random.randint(1, player.toughness)
+        defense = random.randint(1, 18-player.speed)
+
+        player.hit_points -= defense
+        player.enemy_hit_points -= attack
+
+        session.add(player)
+        session.commit()
+        combatant = '{:30s} {:2d}'
+        content = [
+            '```',
+            combatant.format('{}'.format(player.user_name), player.hit_points),
+            combatant.format('Monster', player.enemy_hit_points),
+            '```'
+        ]
+        if player.user_id in self.battles:
+            message = self.battles[player.user_id]
+            message = await self.bot.edit_message(message, '\n'.join(content))
+        else:
+            message = await self.bot.say('\n'.join(content))
+        self.battles[player.user_id] = message

```

Comment:
What is this line doing?

Seems like updating the object's `battles` isn't going to do much - as in it doesn't get stored in the db so it's gonna get wiped, best case, whenever the bot is restarted?

Reply:
That's just the battle message, so that it can be edited, instead of posting a new one. The actual battle values are just the hitpoints, which are stored in the db. If the bot is restarted, the battle will just continue in a new message.